docs/workflow-pod-security-context.md

```diff
@@ -24,5 +24,7 @@ You can configure this globally using [workflow defaults](default-workflow-specs
 !!! Warning "It is easy to make a workflow need root unintentionally"
     You may find that user's workflows have been written to require root with seemingly innocuous code. E.g. `mkdir /my-dir` would require root.
 
-!!! Note "You must use volumes for output artifacts"
-    If you use `runAsNonRoot` - you cannot have output artifacts on base layer (e.g. `/tmp`). You must use a volume (e.g. [empty dir](empty-dir.md)).
+!!! Note "You must use volumes for output artifacts (v3.3 or earlier)"
+    If you use `runAsNonRoot` in versions v3.3 or earlier, you cannot have output artifacts on base layer (e.g. `/tmp`). You must use a volume (e.g. [empty dir](empty-dir.md)).
+    In versions later than v3.3, the [Emissary executor](workflow-executors.md#emissary-emissary)
+    allows artifacts on the base layer with `runAsNonRoot`.
```

Comment:
```suggestion
!!! Note "You must use volumes for output artifacts (v3.3 or earlier with non-emissary executor)"
    If you use `runAsNonRoot` in versions v3.3 or earlier with non-emissary executor, you cannot have output artifacts on base layer (e.g. `/tmp`). You must use a volume (e.g. [empty dir](empty-dir.md)).
    The [Emissary executor](workflow-executors.md#emissary-emissary) (available from v3.0, default since v3.3)
    allows artifacts on the base layer with `runAsNonRoot`.
```

Let's be more precise with the wording.
Otherwise, we could remove this line since <= v3.4 is no longer maintained

Reply:
For all supported versions you don't need volumes it seems, so the whole note could just be removed?

Reply:
When I originally submitted the PR, I didn't know the docs were versioned or that 3.3 was no longer supported.

With that knowledge, if it were my project, I would also lean toward removing the note completely. 

The purpose for submitting the PR in the first place was hopefully to save people time from going down an wrong path.

Reply:
Should I just apply @tczhao 's suggestion, or should I remove the note completely?

Reply:
@paulcwatts Let's remove it then

Reply:
OK, removed!